Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -93,5 +95,146 @@ public static class CustomEntitySearch
+                int cmd = 0;
+                KMPTable[0] = -1;
+
+                while (pos < checkMatch.Length)
+                {
+                    if (wordCharArray[pos] == wordCharArray[cmd])
+                    {
+                        KMPTable[pos] = KMPTable[cmd];
+                    }
+                    else
+                    {
+                        KMPTable[pos] = cmd;
+                        cmd = KMPTable[cmd];
+                        while (cmd >= 0 && wordCharArray[pos] != wordCharArray[cmd])
+                        {
+                            cmd = KMPTable[cmd];
+                        }
+                    }
+                    pos++;
+                    cmd++;
+                }
+                KMPTable[pos] = cmd;
+
+                // Begin searching!
+                int currTextChar = 0;
+                int currWordChar = 0;
+                int offset = 0;
+                StringWriter wordFound = new StringWriter();
+                double currMismatch = 0;
+                IList<char> textCharArray = text.ToCharArray();
+
+                while (currTextChar < textCharArray.Count)
+                {
+                    if(wordCharArray[currWordChar] == textCharArray[currTextChar])
+                    {
+                        wordFound.Write(textCharArray[currTextChar]);
+                        currTextChar++;
+                        currWordChar++;
+
+                        if (currWordChar >= wordCharArray.Count)
+                        {
+                            if (currMismatch < leniency && offset > 0)
+                            {
+                                entities.Add(
+                                    new Entity
+                                    {
+                                        Category = "customEntity",
+                                        Value = wordFound.ToString(),
+                                        Offset = currTextChar - (currWordChar + offset),
+                                        Confidence = (currMismatch / leniency)
+                                    });
+                                addWord = true;
+                                currWordChar = KMPTable[currWordChar];
+                                wordFound.Flush();
+                                currMismatch = 0;
+                                offset = 0;
+                            }
+                            else
+                            {
+                                currWordChar = wordCharArray.Count - 1;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        // fuzzy situation?
+                        if (currMismatch < leniency)
+                        {
+                            string accents = @"[\p{Mn}\p{Mc}]";
+                            currMismatch += Regex.IsMatch(wordCharArray[currWordChar].ToString(), accents) ? 0.5 : 1;
```

Comment:
Also, a comment here would help understand what that regular expression is doing. The Unicode stuff is rather uncommon, and knowledge of how Unicode encodes accented characters is not that widespread. You might want to link to the Unicode spec or some similar resource. It's also very unclear how that's supposed to work. The way I read it, it seems to say that if a character is a mismatch, we only count it as half a mismatch if it is a diacritic/accented character? But that means that a 'ñ' for instance would count as half a mismatch even if the character in the word is 'e', which is completely unrelated.

Also, are you sure this is true of all accented characters? Take 'é': `Regex.IsMatch('é'.ToString(), @"[\p{Mn}\p{Mc}]")` is false.